### PR TITLE
Heading component

### PR DESCRIPTION
**Pull request by azmat239**

### CODE REVIEW

**Comment by UmmarHamid** [comment]
@azmat239 This PR has previous commits from Text components, It should not be the case.
can you fix that and only include relevant commits or need any help?

**Comment by azmat239** [comment]
Need help

File: app/components/Heading/Heading.tsx [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 
-const Heading = ({ text, variant, extraClasses }: HeadingType) => {
-  let tag1 = classNames({
+const Heading: FC<HeadingType> = ({
+  text,
+  variant,
+  extraClasses,
+  as: Tag = "h5",
+}) => {
+  let cls = classNames({
```
Comment (UmmarHamid) [inline app/components/Heading/Heading.tsx:12]: Lets name this as headingClasses

File: app/components/Heading/Heading.tsx [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
     "text-[48px] leading-[48px]": variant == "48",
     "text-[36px] leading-[44px]": variant == "36",
     "text-[32px] leading-[40px]": variant == "32",
     "text-[24px] leading-[30px]": variant == "24",
     "text-[22px] leading-[28px]": variant == "22",
     "text-[20px] leading-[24px]": variant == "20",
   });
-  return <h1 className={`${tag1} ${extraClasses}`}>{text} </h1>;
+
+  return <Tag className={`${cls} ${extraClasses}`}>{text} </Tag>;
```
Comment (UmmarHamid) [inline app/components/Heading/Heading.tsx:21]: Can we implement the logic in a way that if variant 48 is created, the component auto assigns h1?
h2 for 36 and so on.
We won't need the as prop then.

**Review by UmmarHamid — CHANGES_REQUESTED** [review]

**Pull request closed without merging**